ext/web/06_streams.js

```diff
@@ -1019,112 +1193,629 @@
+   */
+  function readableByteStreamControllerShiftPendingPullInto(controller) {
+    assert(controller[_byobRequest] === null);
+    return ArrayPrototypeShift(controller[_pendingPullIntos]);
+  }
+
+  /**
+   * @param {ReadableByteStreamController} controller
+   * @param {PullIntoDescriptor} pullIntoDescriptor
+   * @returns {boolean}
+   */
+  function readableByteStreamControllerFillPullIntoDescriptorFromQueue(
+    controller,
+    pullIntoDescriptor,
+  ) {
+    const elementSize = pullIntoDescriptor.elementSize;
+    const currentAlignedBytes = pullIntoDescriptor.bytesFilled -
+      (pullIntoDescriptor.bytesFilled % elementSize);
+    const maxBytesToCopy = MathMin(
+      controller[_queueTotalSize],
+      pullIntoDescriptor.byteLength - pullIntoDescriptor.bytesFilled,
+    );
+    const maxBytesFilled = pullIntoDescriptor.bytesFilled + maxBytesToCopy;
+    const maxAlignedBytes = maxBytesFilled - (maxBytesFilled % elementSize);
+    let totalBytesToCopyRemaining = maxBytesToCopy;
+    let ready = false;
+    if (maxAlignedBytes > currentAlignedBytes) {
+      totalBytesToCopyRemaining = maxAlignedBytes -
+        pullIntoDescriptor.bytesFilled;
+      ready = true;
+    }
+    const queue = controller[_queue];
+    while (totalBytesToCopyRemaining > 0) {
+      const headOfQueue = queue[0];
+      const bytesToCopy = MathMin(
+        totalBytesToCopyRemaining,
+        headOfQueue.byteLength,
+      );
+      const destStart = pullIntoDescriptor.byteOffset +
+        pullIntoDescriptor.bytesFilled;
+
+      const destBuffer = new Uint8Array(
+        pullIntoDescriptor.buffer,
+        destStart,
+        bytesToCopy,
+      );
+      const srcBuffer = new Uint8Array(
+        headOfQueue.buffer,
+        headOfQueue.byteOffset,
+        bytesToCopy,
+      );
+      destBuffer.set(srcBuffer);
+
+      if (headOfQueue.byteLength === bytesToCopy) {
+        ArrayPrototypeShift(queue);
+      } else {
+        headOfQueue.byteOffset += bytesToCopy;
+        headOfQueue.byteLength -= bytesToCopy;
+      }
+      controller[_queueTotalSize] -= bytesToCopy;
+      readableByteStreamControllerFillHeadPullIntoDescriptor(
+        controller,
+        bytesToCopy,
+        pullIntoDescriptor,
+      );
+      totalBytesToCopyRemaining -= bytesToCopy;
+    }
+    if (!ready) {
+      assert(controller[_queueTotalSize] === 0);
+      assert(pullIntoDescriptor.bytesFilled > 0);
+      assert(pullIntoDescriptor.bytesFilled < pullIntoDescriptor.elementSize);
+    }
+    return ready;
+  }
+
+  /**
+   * @param {ReadableByteStreamController} controller
+   * @param {number} size
+   * @param {PullIntoDescriptor} pullIntoDescriptor
+   * @returns {void}
+   */
+  function readableByteStreamControllerFillHeadPullIntoDescriptor(
+    controller,
+    size,
+    pullIntoDescriptor,
+  ) {
+    assert(
+      controller[_pendingPullIntos].length === 0 ||
+        controller[_pendingPullIntos][0] === pullIntoDescriptor, // TODO: check
```

Comment:
Looks correct. You doing a reference equality check here, not a value equality check, which looks to be what the spec intends.